src/zenml/utils/artifact_utils.py

```diff
@@ -243,30 +280,53 @@ def load_artifact_visualization(
 
     try:
         artifact_store = cast(
-            BaseArtifactStore, StackComponent.from_model(artifact_store_model)
+            "BaseArtifactStore",
+            StackComponent.from_model(artifact_store_model),
         )
     except ImportError:
         link = "https://docs.zenml.io/component-gallery/artifact-stores/custom#enabling-artifact-visualizations-with-custom-artifact-stores"
         raise NotImplementedError(
-            f"Artifact '{artifact.id}' could not be visualized because the "
+            f"Artifact '{artifact.id}' could not be loaded because the "
             f"underlying artifact store '{artifact_store_model.name}' "
             f"could not be instantiated. This is likely because the "
             f"artifact store's dependencies are not installed. For more "
             f"information, see {link}."
         )
+    return artifact_store
 
-    visualization = artifact.visualizations[index]
-    mode = "rb" if visualization.type == VisualizationType.IMAGE else "r"
+
+def _load_uri_from_artifact_store(
+    uri: str,
+    artifact_store: "BaseArtifactStore",
+    mode: str = "rb",
+) -> Any:
+    """Load the given uri from the given artifact store.
+
+    Args:
+        uri: The uri of the file to load.
+        artifact_store: The artifact store to load the file from.
+        mode: The mode to open the file in.
+
+    Returns:
+        The loaded file.
+
+    Raises:
+        DoesNotExistException: If the file does not exist in the artifact store.
+        NotImplementedError: If the artifact store cannot open the file.
+    """
     try:
-        with artifact_store.open(visualization.uri, mode) as text_file:
-            value = text_file.read()
+        with artifact_store.open(uri, mode) as text_file:
+            return text_file.read()
     except FileNotFoundError:
         raise DoesNotExistException(
-            f"Artifact '{artifact.id}' could not be visualized because the "
-            f"visualization file '{visualization.uri}' was not be found in "
-            f"the underlying artifact store '{artifact_store_model.name}'."
+            f"File '{uri}' does not exist in artifact store "
+            f"'{artifact_store.name}'."
+        )
+    except Exception:
+        link = "https://docs.zenml.io/component-gallery/artifact-stores/custom#enabling-artifact-visualizations-with-custom-artifact-stores"
```

Comment:
Could you also please do a logger.exception() here so we don't lose the error info ? catching generic exceptions and then ignoring their source can be devastating for maintainability.

Reply:
Done 👍